utils.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2020 and later: Google, Inc.
+
+import os
+import random
+import shutil
+
+def _check_create_dir(dir_path):
+    """Check if the given path exists and create if not.
+
+    Args: dir: Str, path to directory.
+
+    Raises:
+        OSError: if specified directory cannot be created."""
+    # Check if out_dir exists and create
+    if not os.path.isdir(dir_path):
+        print("{} does not exist, creating directory..."
+              .format(dir_path))
+        try:
+            os.mkdir(dir_path)
+        except OSError:
+            print("Creation of directory {} failed."
+                  .format(dir_path))
+            raise
+        else:
+            print("New directory successfully created.")
+
+def train_test_split(train_dir, test_dir, num_test=100):
+    """Split dataset (already created) into training and testing datasets.
+    Expect all data record to be in train_dir. Expect test_dir to be either
+    empty or does not exists.
+
+
+    Args:
+        train_dir: Str, relative path to the directory containing training
+            images.
+        test_dir: Str, relative path to test directory (expect to be non-exist
+            or emtpy).
+        num_test: Int, number of test records.
+
+    Returns:
+        num_train: Int, total number of training records.
+        num_test: Int, total number of test records.
+
+    Raises:
+        OSError: If no images found in out_dir.
+        ValueError: If num_test is larger than total number of records.
+        OSError: If test data already exists.
+    """
+    # Get absolute path to train and test data directory
+    train_dir_abs = os.path.join(os.getcwd(), train_dir)
+    test_dir_abs = os.path.join(os.getcwd(), test_dir)
+
+    # Create test dir
+    _check_create_dir(test_dir_abs)
+
+    # Get total number of training records
+    num_total = len([name for name in os.listdir(train_dir_abs)
+                     if os.path.isfile(os.path.join(train_dir_abs, name))])
+    num_exist = len([name for name in os.listdir(test_dir_abs)
```

Comment:
num_exist  = len(next(os.walk(test_dir_abs))[2])
see https://stackoverflow.com/questions/2632205/how-to-count-the-number-of-files-in-a-directory-using-python